schematic/store/synapse.py

```diff
@@ -1119,6 +1119,13 @@ def _get_file_entityIds(
                     "No manifest was passed in, a manifest is required when `only_new_files` is True."
                 )
 
+            if "entityId" not in manifest.columns:
```

Comment:
Just because I'm a worry-wort, can you tell me when the `_get_file_entityIds` function is called?  Otherwise, this is great!

Reply:
It is called when:
- `add_annotations_to_entities_files` is called - not relevant here because it is called with `only_new_files=False`
- `_get_files_metadata_from_dataset` is called - not relevant here because it is called with `only_new_files=False`
- `fill_in_entity_id_filename` is called - twice, once with `only_new_files=True`
     - This is in turn called when a file-based manifest is updated (`updateDatasetManifestFiles`)

That being said, I don't foresee any unintended consequences (famous last words) to this change because it is the line before a `KeyError` will always be raised if the manifest doesn't have that column.

Reply:
@GiaJordan Hmmm, so what you're saying is there's two scenarios that people may hit this issue:
1. generating an empty manifest when there's a manifest in the dataset folder
2. generating a manifest when there's a manifest in the dataset folder and we want annotations filled?

Reply:
Oops. accidentally resolved it... @GiaJordan

Reply:
Yes, schematic always looks to see if a manifest exists in the dataset (if an id is provided) when generating a manifest. In this case because a manifest exists but its formatted incorrectly it would cause issues when determining how to order the existing/new files that exist on the dataset.

Reply:
> unless our recommendation of generating an empty manifest will fail.

This is what I'm saying will happen

> Correct me if I'm wrong, but if creation of a new blank manifest is unable to occur, that was already true before this change.

No you're correct, this is about the recommended suggestion to get users back into the flow, now raising the exception itself

Reply:
Yikes. We definitely don't want creation of an empty manifest to fail if there's an existing manifest. 

So then if we add this, will we need that error message?

Reply:
> what happens when submitting a manifest without the entityId column? That gets filled in if the Filename is correct right?

No, entityId are required to annotate the files. There are some cases where the ids are present and then erroneously dropped, but there are other cases where they weren't present in the first place, like this one. When submitting the raised error is the same iirc @thomasyu888

Reply:
We will need to test both of these solutions for when there isn't a manifest in the folder
This is for when there is an existing manifest in the folder
```schematic manifest -c test_flat_config.yml get -dt BulkRNA-seqAssay -d syn64109617 -s```
^ This will also cause the same error, so we don't want this to error and we want an empty manifest returned (ideally with entityId) filled out

```schematic manifest -c test_flat_config.yml get -dt BulkRNA-seqAssay -d syn64109617 -s -a```
^ We want to throw an error

OR

Product research done on asking if people are ok with blank manifests returned even with the `-a` parameter (if entityId) is NOT present.

Reply:
I was messing around with this and I think the simplest way to trigger the new manifest to generate is just to add a check for the `entityId` column in `SynapseStorage.updateDatasetManifestFiles` directly instead of doing some error handling which is a bit more complicated due to the nested nature of the functions. I added this in my latest [commit](https://github.com/Sage-Bionetworks/schematic/pull/1551/commits/6795c4cb947936ff58b2181e123f4e4420a23b44). Thoughts @GiaJordan @thomasyu888? Any consequences of doing this I haven't thought of?